src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -302,7 +304,7 @@ private String getNextTaskType(String taskType) {
     }
     if (taskType.equals(HDFSConstants.JOURNAL_NODE_ID)) {
       for (int i = 1; i <= conf.getJournalNodeCount(); i++) {
-        if (!liveState.getJournalNodeNames().containsValue(HDFSConstants.JOURNAL_NODE_ID + i)) {
+        if (!journalNodeTaskNames.contains(HDFSConstants.JOURNAL_NODE_ID + i)) {
           return HDFSConstants.JOURNAL_NODE_ID + i;
```

Comment:
Since we're never supposed to reach here when we've launched enough tasks, lets return an error instead of ""